### DISCUSSION TITLE
Difference between useActionData() and useFetcher().data?

### DISCUSSION THREAD

**Discussion by armandAkop** [remix-run/remix#5086] [Q&A] [answered]
I'm reading the docs and I'm a little confused about the differences between useActionData()

> This hook returns the JSON parsed data from your route action. It returns undefined if there hasn't been a submission at the current location yet.

And useFetcher().data

> The returned response data from your loader or action is stored here. Once the data is set, it persists on the fetcher even through reloads and resubmissions (like calling fetcher.load() again after having already read the data).

1. If I had submitted a form via fetcher.submit() to a route that return JSON data, wouldn't useActionData() and useFetcher().data refer to the same data?
2. For useFetcher().data - "The returned response data **from your loader** or action is stored here" - is that referring to the data if I were to call fetcher.load() somewhere on my page?

**Comment by sergiodxa** [comment, marked as answer]
If you submit to an action with a fetcher then only fetcher.data will contain the result.

If you used Form or useSubmit then the result will be in useActionData.

**Reply by armandAkop** [reply]
Thank you! Makes sense now.
